### PR TITLE
T23859 Initial library implementation

### PR DESCRIPTION
**Pull request by pwithnall**
Implements everything from https://phabricator.endlessm.com/T23859 except Debian packaging, which I’ll work on shortly. Tested and ready for review otherwise though.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -29,4 +29,76 @@ polkit_gobject = dependency('polkit-gobject-1')
...
+  '-Wswitch-default',
+  '-Wswitch-enum',
+  '-Wtype-limits',
+  '-Wundef',
+  '-Wuninitialized',
+  '-Wunused-but-set-variable',
+  '-Wunused-result',
+  '-Wunused-variable',
+  '-Wwrite-strings'
+]
```
Comment (GeorgesStavracas) [inline meson.build:102]: I think it's easier to just use `-Wall -Wextra -Weverything` if we want all those warnings.

**Review by GeorgesStavracas — APPROVED** [review]
Left only one minor comment, but otherwise looks (very) good to me. Feel free to merge after addressing that comment.

File: meson.build (hunk 1 above)
Comment (wjt) [reply meson.build:102]: See https://github.com/mesonbuild/meson/issues/2609 for relevant discussion.

Comment (pwithnall) [reply meson.build:102]: Not sure `-Weverything` gives the right curated set of warnings. Not fussed enough about going back into that argument to change this now.

**Pull request merged**